third_party/gpus/cuda_configure.bzl

```diff
@@ -316,11 +316,36 @@ def _get_cxx_inc_directories_impl(repository_ctx, cc, lang_is_cpp, tf_sysroot):
         ).stdout.strip() + "/share"
         inc_dirs += "\n" + resource_dir
 
-    return [
+    compiler_includes = [
         _normalize_include_path(repository_ctx, _cxx_inc_convert(p))
         for p in inc_dirs.split("\n")
     ]
 
+    # fix include path by also including paths where resolved symlink is replaced by original path
```

Comment:
This is really confusing. It's in line with the rest of this function, which is also really confusing, so that's not really a deal breaker.

I think it would be ok to include this if you extend the comment with hard examples of what's going on, like the possible results that cc might return and what this chunk of code will do to them.

Also @perfinion may have some requests here.

Reply:
> I think it would be ok to include this if you extend the comment with hard examples of what's going on, like the possible results that cc might return and what this chunk of code will do to them.

I extended the introductory comment and added examples further below.
In anticipation of a potential objection about duplicate paths being added I added a check in the construction of `original_compiler_includes`

Does that suffice?

Reply:
Thanks for adding more details. Does this work for clang as well?